nomad/state/state_store.go

```diff
@@ -1710,6 +1710,55 @@ func (s *StateStore) CSIVolumesByIDPrefix(ws memdb.WatchSet, namespace, volumeID
 	return wrap, nil
 }
 
+// CSIVolumesByNodeID looks up CSIVolumes in use on a node
+func (s *StateStore) CSIVolumesByNodeID(ws memdb.WatchSet, nodeID string) (memdb.ResultIterator, error) {
+	allocs, err := s.AllocsByNode(ws, nodeID)
+	if err != nil {
+		return nil, fmt.Errorf("alloc lookup failed: %v", err)
+	}
+	snap, err := s.Snapshot()
+	if err != nil {
+		return nil, fmt.Errorf("snapshot failed: %v", err)
+	}
+
+	allocs, err = snap.DenormalizeAllocationSlice(allocs)
+	if err != nil {
+		return nil, fmt.Errorf("alloc denormalize failed: %v", err)
```

Comment:
Is this error message meaningful to operators?

Reply:
Fixed in 666a8a7f8